crates/amaru/src/ledger/state/transaction.rs

```diff
@@ -51,10 +52,21 @@ pub fn apply<T>(
         .certificates
         .map(|xs| xs.to_vec())
         .unwrap_or_default();
-
     span.record("transaction.certificates", certificates.len());
     apply_certificates(&span, &mut state.pools, &mut state.accounts, certificates);
 
+    let withdrawals = transaction_body
+        .withdrawals
+        .unwrap_or_else(|| NonEmptyKeyValuePairs::Def(vec![]));
+    span.record("transaction.withdrawals", withdrawals.len());
+    state
+        .withdrawals
+        .extend(withdrawals.iter().map(|(account, _)| {
+            reward_account_to_stake_credential(account).unwrap_or_else(|| {
```

Comment:
Could be replaced with `expect`?
Now I'm wondering if we want to fail in those cases? And what happens if a panic is thrown here?